science/paraview/Portfile

```diff
@@ -110,25 +107,53 @@ destroot {
 
 
 # other pythons?
-variant python27 conflicts python36 description {Add Python 2.7 support.} {
+variant python27 conflicts python36 python37 description {Add Python 2.7 support.} {
     depends_lib-append  port:py27-matplotlib
     configure.args-append \
         -DPARAVIEW_ENABLE_PYTHON:BOOL=ON \
         -DPYTHON_EXECUTABLE=${prefix}/bin/python2.7 \
         -DPYTHON_INCLUDE_DIR=${frameworks_dir}/Python.framework/Versions/2.7/Headers/ \
         -DPYTHON_LIBRARY=${prefix}/lib/libpython2.7.dylib
+
+    if {[mpi_variant_isset]} {
+        depends_lib-append port:py27-mpi4py
+        configure.args-append \
+            -DVTK_USE_SYSTEM_MPI4PY:BOOL=ON
+    }
 }
-variant python36 conflicts python27 description {Add Python 3.6 support.} {
+variant python36 conflicts python27 python37 description {Add Python 3.6 support.} {
     depends_lib-append  port:py36-matplotlib
     configure.args-append \
         -DPARAVIEW_ENABLE_PYTHON:BOOL=ON \
         -DPYTHON_EXECUTABLE=${prefix}/bin/python3.6 \
         -DPYTHON_INCLUDE_DIR=${frameworks_dir}/Python.framework/Versions/3.6/Headers/ \
         -DPYTHON_LIBRARY=${frameworks_dir}/Python.framework/Versions/3.6/Python \
         -DVTK_PYTHON_VERSION=3
+
+    if {[mpi_variant_isset]} {
+        depends_lib-append port:py36-mpi4py
+        configure.args-append \
+            -DVTK_USE_SYSTEM_MPI4PY:BOOL=ON
+    }
+}
+
+variant python37 conflicts python27 python36 description {Add Python 3.7 support.} {
```

Comment:
This is outside the scope of this PR, but there is a lot of duplicated boilplate code here, defining each of the python variants. In fact, each only really differs in the version number, so it would be nice to clean this up by using a `foreach` loop to define the variants. Would making adding new ones later on as easy as just adding a version to a list...

Reply:
Ok. I'm not a Tcl ninja that's why I opted for a boilerplate version. :). I'll fix as you suggested

Reply:
@rubendibattista Have you been able to make the requested changes ?

Reply:
Not yet. There's also a [patch to be applied regarding the wrong render of an icon](https://gitlab.kitware.com/paraview/paraview/merge_requests/3150)... 

I'll try to finalize this week.

Reply:
```
# Supported pythons
set python_versions {27 35 36 37}

foreach pyver ${python_versions} {
    # Conflicting python versions
    set other_python_versions {}
    foreach other_pyver ${python_versions} {
        if {${other_pyver} ne ${pyver}} {
            if {${other_pyver} ni ${other_python_versions}} {
                lappend other_python_versions python${other_pyver}
                }
            }
        }

    # Get python branch
    set python_branch {[string range ${pyver} 0 end-1].[string index ${pyver} end]}

    variant python${pyver} conflicts [join ${other_python_versions} " "] {Add Python ${python_branch} support.} {
            depends_lib-append port:py${pyver}-matplotlib
            configure.args-append \
                -DPARAVIEW_ENABLE_PYTHON:BOOL=ON \
                -DPYTHON_EXECUTABLE=${prefix}/bin/python${python_branch} \
                -DPYTHON_INCLUDE_DIR=${frameworks_dir}/Python.framework/Versions/${python_branch}/Headers/ \
                -DPYTHON_LIBRARY=${prefix}/lib/libpython${python_branch}.dylib

            if {[mpi_variant_isset]} {
                depends_lib-append port:py${pyver}-mpi4py
                configure.args-append \
                    -DVTK_USE_SYSTEM_MPI4PY:BOOL=ON
            }
    }
    
}

```
This is what I came out with. Not a Tcl ninja so sorry if it's very ugly and there's a better way. The only thing is I'm getting this Warning:
```
--->  Verifying Portfile for paraview
Warning: Variant python27 conflicts with non-existing variant python35 python36 python37
Warning: Variant python27 conflicts with non-existing variant Add Python ${python_branch} support.
Warning: Variant python35 conflicts with non-existing variant python27 python36 python37
Warning: Variant python35 conflicts with non-existing variant Add Python ${python_branch} support.
Warning: Variant python36 conflicts with non-existing variant python27 python35 python37
Warning: Variant python36 conflicts with non-existing variant Add Python ${python_branch} support.
Warning: Variant python37 conflicts with non-existing variant python27 python35 python36
Warning: Variant python37 conflicts with non-existing variant Add Python ${python_branch} support.
--->  0 errors and 8 warnings found.
```

The conflicting variants look correct...